lib/mongoid/lazy_migration/document.rb

```diff
@@ -63,8 +63,7 @@ def try_lock_migration
     selector = atomic_selector.merge('migration_state' => { "$in" => [nil, 'pending'] })
     changes  = { "$set" => { 'migration_state' => 'processing' }}
     safety   = { :safe => true }
-
-    self.class.with(safety).where(selector).query.update(changes)['updatedExisting']
+    self.class.with(safety).where(selector).update(changes).successful?
```

Comment:
I think `update` only works on an existing object. If you want to create a new object, you have to `upsert`.

https://docs.mongodb.org/ecosystem/tutorial/mongoid-persistence/